### DISCUSSION TITLE
Proper recursive ls function

### DISCUSSION THREAD

**Discussion by dch82** [nushell/nushell#9768] [Show and tell]
This function recursively ls's into a tree of tables:
```
# ls -r replacement
# Recommended usage: pipe into `table -e`
def lsr [
  dir = "."         # Directory to start in
] {
  cd $dir
  ls | select name | merge (ls | get name | par-each { |item| cd $item; lsr } | wrap contents)                                
}
```
You can find it in my project [nutils](https://github.com/dch82/nutils).

**EDIT 1** Parallelised the function

**Comment by fdncred** [comment]
You can probably make it a quite a bit faster by changing `each` to `par-each`.

In my nushell repo folder it takes it from `8sec 53ms 344µs` to `877ms 893µs 100ns`

**Comment by dch82** [comment]
@fdncred It's been added!
